services/chat/api.go

```diff
@@ -408,14 +435,11 @@ func (api *API) getChatAndCommunity(pubKey string, communityID types.HexBytes, c
 
 	var community *communities.Community
 	if messengerChat.CommunityID != "" {
-		communityID, err := hexutil.Decode(messengerChat.CommunityID)
-		if err != nil {
-			return nil, nil, err
-		}
+		var err error
+		community, err = api.getCommunityByID(messengerChat.CommunityID)
 
-		community, err = api.s.messenger.GetCommunityByID(communityID)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, ErrCommunityNotFound
```

Comment:
I think we should only return `ErrCommunityNotFound` if we know for sure that was the issue. I think that's what @cammellos would say :laughing:

Reply:
Refactored to return `ErrCommunityNotFound` only if we know the community was not found. For other cases return an occurred error

P.S. Hmmm... maybe we should return this error during the DB request...